src/platform/site-wide/user-nav/containers/Main.jsx

```diff
@@ -164,9 +164,10 @@ export class Main extends React.Component {
 
 export const mapStateToProps = state => {
   const { form = {} } = state;
-  const { formAutoSavedStatus = '', additionalRoutes = [] } = form;
+  const { formAutoSavedStatus, additionalRoutes = [] } = form;
   const additionalSafePaths = additionalRoutes.map(route => route.path);
   const shouldConfirmLeavingForm =
+    typeof formAutoSavedStatus !== 'undefined' &&
```

Comment:
Maybe try this:

```
const { form } = state;
const additionalSafePaths = form && form.additionalRoutes.map(route => route.path)
const shouldConfirmLeavingForm =
  form &&
  form.formAutoSavedStatus !== SAVE_STATUSES.success &&
  isInProgressPath(window.location.pathname, additionalSafePaths);
```

Reply:
Ah, I made a change before seeing this comment. Check it out my latest commit.l

Reply:
I like yours better. will follow up again.

Reply:
actually I'll leave mine as written.